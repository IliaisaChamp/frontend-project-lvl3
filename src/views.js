import * as _ from 'lodash';

const addModalInfo = (data) => {
  const buttons = document.querySelectorAll('[data-bs-target="#modal"]');

  buttons.forEach((button) => {
    button.addEventListener('click', (e) => {
      const { id } = e.target.dataset;
      const { url } = e.target.dataset;
      const { title, description, link } = data[url].posts[id - 1];
      const modal = document.querySelector('#modal');
      modal.querySelector('.modal-title').textContent = title;
      modal.querySelector('.modal-body').textContent = description;
      modal.querySelector('a').href = link;
    });
  });
};

const addArticlesToList = (posts, url) => {
  const createArticle = (post, id) => {
    const link = `<a href="${post.link}" class="font-weight-bold" data-id="" target="_blank" rel="noopener noreferrer">${post.title}</a>`;
    const button = `<button type="button" class="btn btn-primary btn-sm" data-id="${id + 1}" data-url="${url}" data-bs-toggle="modal" data-bs-target="#modal">Просмотр</button>`;
    return `<li class="list-group-item d-flex justify-content-between align-items-start">${link}${button}</li>`;
  };

  const listItems = posts.map(createArticle);
  return listItems.join('');
};

const addFeedsToList = (feeds) => {
  const { title, description } = feeds;
  return `<li class="list-group-item"><h3>${title}</h3><p>${description}</p></li>`;
};

const createFeed = (data) => {
  const keys = _.keys(data);
  keys.reverse();
  const feedsContainer = document.querySelector('.feeds');
  const postsContainer = document.querySelector('.posts');

  feedsContainer.innerHTML = `<h2>Фиды</h2><ul class="list-group mb-5">${keys
    .map((url) => addFeedsToList(data[url]))
    .join('')}</ul>`;
  postsContainer.innerHTML = `<h2>Посты</h2><ul class="list-group mb-5">${keys
    .map((url) => addArticlesToList(data[url].posts, url))
    .join('')}</ul>`;
  addModalInfo(data);
};

const updateFeed = (data) => {
  createFeed(data);
};

export { createFeed, updateFeed, addFeedsToList };
