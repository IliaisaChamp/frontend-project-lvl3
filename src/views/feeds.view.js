import state from '../state';

const makeFeedList = () => {
  let lists = '';
  state.feeds.feedsList.forEach((list) => {
    lists += `
      <li class="list-group-item">
        <h3>${list.title}</h3>
        <p>${list.description}</p>
        </li>
    `;
  });
  return lists;
};
const makePostList = () => {
  let list = '';
  state.feeds.posts.forEach((postsArr) => {
    postsArr.forEach((item, id) => {
      list += `
      <li class="list-group-item d-flex justify-content-between align-items-start">
        <a href="${item.querySelector('link').textContent}" class="font-weight-bold" 
        data-id="${id + 1}" target="_blank" rel="noopener noreferrer">${item.querySelector('title').textContent}</a>
        <button type="button" class="btn btn-primary btn-sm" data-id="${id + 1}" data-toggle="modal" data-target="#modal">Просмотр</button>
      </li>
      `;
    });
  });
  return list;
};

export default () => {
  const feeds = document.querySelector('.feeds');
  const posts = document.querySelector('.posts');

  feeds.innerHTML = `
    <h2>Фиды</h2>
    <ul class="list-group mb-5">
      ${makeFeedList()}
    </ul>
  `;

  posts.innerHTML = `
    <h2>Посты</h2>
    <ul class="list-group">
      ${makePostList()}
    </ul>
  `;
};
