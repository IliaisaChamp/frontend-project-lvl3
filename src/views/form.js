const formHTML = `
  <form action="" class="rss-form"><div class="form-row">
    <div class="col">
      <input autofocus="" required="" name="url" aria-label="url" class="form-control form-control-lg w-100" placeholder="ссылка RSS"></div>
      <div class="col-auto">
        <button type="submit" aria-label="add" class="btn btn-lg btn-primary px-sm-5">Add</button>
      </div>
    </div>
  </form>
`;

export default () => formHTML;
