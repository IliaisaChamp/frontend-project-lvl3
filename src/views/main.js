import form from './form';

const mainHtml = `
<main class="flex-grow-1">
  <section class="container-fluid bg-secondary p-5">
    <div class="row">
      <div class="col-md-10 col-lg-8 mx-auto text-white">
        <h1 class="display-3">RSS агрегатор</h1>
        ${form()}
        <p class="text-light my-1">Пример: https://ru.hexlet.io/lessons.rss</p>
      </div>
    </div>
  </section>
</main>
`;
git
export default () => mainHtml;
