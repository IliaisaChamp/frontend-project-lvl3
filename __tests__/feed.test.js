import { addFeedsToList } from '../src/views.js';

const data = {
  title: 'Test',
  description: 'test desc',
};

it('renders correctly', () => {
  expect(addFeedsToList(data)).toMatchSnapshot();
});
