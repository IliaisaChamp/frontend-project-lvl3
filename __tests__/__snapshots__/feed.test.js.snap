// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `"<li class=\\"list-group-item\\"><h3>Test</h3><p>test desc</p></li>"`;
